invoice/src/main/resources/org/killbill/billing/invoice/ddl.sql

```diff
@@ -236,3 +236,16 @@ CREATE UNIQUE INDEX invoice_parent_children_id ON invoice_parent_children(id);
 CREATE INDEX invoice_parent_children_invoice_id ON invoice_parent_children(parent_invoice_id);
 CREATE INDEX invoice_parent_children_tenant_account_record_id ON invoice_parent_children(tenant_record_id, account_record_id);
 CREATE INDEX invoice_parent_children_child_invoice_id ON invoice_parent_children(child_invoice_id);
+
+CREATE TABLE invoice_billing_events (
+    record_id serial unique,
+    id varchar(36) NOT NULL,
+    invoice_id varchar(36) NOT NULL,
+	lz_billing_events blob NOT NULL,
```

Comment:
Nit: don't want to sound pedantic (😏), but I'm not a big fan of the `lz` prefix (here and throughout the code), as it's a bit confusing. It hints that it's compressed, but it isn't clear enough *how* it is compressed.
`ning-compress` is a Java implementation of LibLZF (http://oldhome.schmorp.de/marc/liblzf.html) but there are many other LZ- algorithms (e.g. https://github.com/ning/jvm-compressor-benchmark/wiki).

I would have probably kept the `BillingEventSet` object all the way to the dao, and have the dao in charge of serialization (and name the column `billing_events`).